asm/vm/asm_vm

```diff
@@ -1340,6 +1567,87 @@ EOF
   retry 3 kpt pkg get --auto-set=false "${KPT_URL}" asm
 }
 
+install_expansion_gateway() {
+  info "Installing the expansion gateway..."
+  retry 5 istioctl install -f "${EXPANSION_GATEWAY_FILE}"
+
+  # Prevent the stderr buffer from ^ messing up the terminal output below
+  sleep 1
+  info "...done!"
+}
+
+expose_istiod() {
+  info "Exposing the control plane for VM workloads..."
+  retry 3 kubectl apply -f "${EXPOSE_ISTIOD_SERVICE}"
+
+  ASM_REVISIONS="$(strip_last_char "${ASM_REVISIONS}")"
+  for rev in $(echo "${ASM_REVISIONS}" | tr ',' '\n'); do
+    kpt cfg set asm anthos.servicemesh.istiodHost "istiod-${rev}.istio-system.svc.cluster.local"
+    kpt cfg set asm anthos.servicemesh.istiod-vs-name "istiod-vs-${rev}"
+    kpt cfg set asm anthos.servicemesh.istiod-dr-name "istiod-dr-${rev}"
+  done
+
+  retry 3 kubectl apply -f "${EXPOSE_ISTIOD_SERVICE}"
+}
+
+install_google_identity_provider() {
+  info "Registering GCE Identity Provider in the cluster..."
+  retry 3 kubectl apply -f asm/identity-provider/identityprovider-crd.yaml
+  retry 3 kubectl apply -f asm/identity-provider/googleidp.yaml
+}
+
+enable_service_mesh_feature() {
+  info "Enabling the service mesh feature..."
+
+  local TOKEN
+  TOKEN="$(retry 2 gcloud \
+    --project="${PROJECT_ID}" auth print-access-token)"
+
+  local RESPONSE
+  RESPONSE="$(run curl -s -H "X-Goog-User-Project: ${PROJECT_ID}"  \
+    "https://gkehub.googleapis.com/v1alpha1/projects/${PROJECT_ID}/locations/global/features/servicemesh" \
+    -H @- <<EOF
+Authorization: Bearer ${TOKEN}
+EOF
+)"
+
+  if [[ "$(echo "${RESPONSE}" | jq -r '.featureState.lifecycleState')" == "ENABLED" ]]; then
+    info "The service mesh feature is already enabled."
+  elif [[ "$(echo "${RESPONSE}" | jq -r '.error.code')" -eq 404 ]]; then
+    run curl -s -H "Content-Type: application/json" \
+      -XPOST "https://gkehub.googleapis.com/v1alpha1/projects/${PROJECT_ID}/locations/global/features?feature_id=servicemesh"\
+      -d '{servicemesh_feature_spec: {}}' \
+      -H @- <<EOF
+Authorization: Bearer ${TOKEN}
```

Comment:
Is it possible to fail at this step? We may be able to run GET to see if the feature is not enabled, but we may not have enough permission to actually enable it.

Reply:
I'd like to do this as a follow up to this PR where I will add more validation to the script. The validation will reuse scripts from install_asm.